_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,420 @@
+---
+layout: post
+title:  "Reassigning partitions in Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+As a Kafka user, we sometimes have to scale up/down our kafka brokers depending on the use case. Addition of extra brokers can be an advantage to handle massive load, and we can use Cruise Control for general rebalancing in Strimzi since it allows us to automate the balancing of load across the cluster but what if we are scaling down the clusters. Let us understand this with the help of an example, suppose there are certain number of brokers in a cluster and now we want to remove a broker from the cluster. We need to make sure that the broker which is going to be removed should not have any assigned partitions. Cruise Control currently doesn't support scaling down the cluster. This requires a tool that can assign the partitions from the broker to be removed, to the remaining brokers. The most convenient tool for this job is the Kafka reassignment partition tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here :-
+
+1. You can reassign the partition between the brokers anytime. For eg. it can be used at times when you want to scale down your cluster.
```

Comment:
```suggestion
1. You can reassign the partition between the brokers anytime. For example it can be used at times when you want to scale down your cluster.
```